src/x/test/util.go

```diff
@@ -32,3 +32,17 @@ func ByteSlicesBackedBySameData(a, b []byte) bool {
 	bHeader := (*reflect.SliceHeader)(unsafe.Pointer(&b))
 	return aHeader.Data == bHeader.Data
 }
+
+// ByteSliceDataContained checks data of 'inner' slice is fully contained inside data of
+// 'outter' slice.
+func ByteSliceDataContained(outter, inner []byte) bool {
+	var (
+		outterHeader = (*reflect.SliceHeader)(unsafe.Pointer(&outter)) //nolint:gosec
+		innerHeader  = (*reflect.SliceHeader)(unsafe.Pointer(&inner))  //nolint:gosec
+		outterStart  = uint(outterHeader.Data)
+		outterEnd    = outterStart + uint(outterHeader.Len)
+		innerStart   = uint(innerHeader.Data)
+		innerEnd     = innerStart + uint(innerHeader.Len)
+	)
+	return outterStart <= innerStart && innerEnd <= outterEnd
```

Comment:
Nit: I believe it should be spelled `outer`.

Reply:
https://github.com/m3db/m3/pull/3075/commits/6bb6d9b40510ae5404955846d12c5319a9dc0a9e#diff-f726ae545aec3259d1475db2af7e4909d09ea1981db15fa3631fca37941956b4L35-L48